src/service/data_products/taxa_count.py

```diff
@@ -232,11 +255,49 @@ async def get_taxa_counts(
     q = await _query(store, collection_id, load_ver, rank)
     for dp_proc in [dp_match, dp_sel]:
         await _add_subset_data_in_place(q, store, collection_id, load_ver, rank, dp_proc)
-        # For now always sort by the std data. See if ppl want sort by match/selection data
-        # before implementing something more sophisticated.
+
+    _sort_taxa_counts(q, sort_priority, [dp_match, dp_sel])
+
     return _taxa_counts(dp_match=dp_match, dp_sel=dp_sel, data=q)
 
 
+def _fill_missing_orders(sort_order: list[str]):
+    # fill in missing orders with the default sort order
+    if not isinstance(sort_order, list):
+        raise ValueError(f"sort_order must be a list of strings, provided: {sort_order}")
+
+    return [order for order in _DEFAULT_SORT_ORDER if order not in sort_order] + sort_order
+
+
+def _sort_taxa_counts(
+        q: list[dict[str, Any]],
+        sort_priority: list[str],
+        dp_list: list[models.DataProductProcess]):
+    # Sort taxa count records in place by the sort_priority list.
+    processed_count = [_TYPE2FIELD[dp.type] for dp in dp_list if dp]
+
+    if processed_count:
+
+        if len(sort_priority) != len(set(sort_priority)):
+            raise errors.IllegalParameterError(f"Duplicate sort priority found: {sort_priority}")
+
+        try:
+            sort_order_rev = [_SORT_PRIORITY_ORDER_MAP[k] for k in sort_priority]
+        except KeyError as e:
+            raise errors.IllegalParameterError(f"Invalid sort priority: {e}, "
+                                               f"valid priorities are: {list(_SORT_PRIORITY_ORDER_MAP.keys())}")
+
+        # fill in missing orders with the default precedence order
+        sort_order = _fill_missing_orders(sort_order_rev[::-1])
+        # remove any sort orders that are not processed (i.e. match or selection counts)
+        sort_order = [order for order in sort_order if order in processed_count or order == names.FLD_TAXA_COUNT_COUNT]
```

Comment:
Assuming the above change, the only thing that can be missing at this point is the regular count, right? So can't you just do something like

```
if count_str not in sort_order_rev:
    sort_order_rev += [count_str]
sort_order = sort_order_rev.reverse()
```

Reply:
if we are doing that, we are not enforcing the default precedence right? For example, if `sort_priority='count'`, we still want to sort records by selected then by matched.

Reply:
Yep, you're right